frontend/ui/elements/screensaver_menu.lua

```diff
@@ -26,114 +26,144 @@ local function genMenuItem(text, setting, value, enabled_func, separator)
+                    genMenuItem(_("Current state"), "screensaver_img_background", "none", nil, true),
+                    -- separator
+                        {
+                            text_func = function()
+                                local percentage = G_reader_settings:readSetting("screensaver_stretch_limit_percentage")
+                                if G_reader_settings:isTrue("screensaver_stretch_images") and percentage then
+                                    return T(_("Stretch to fit screen (with limit: %1 %)"), percentage)
+                                end
+                                return _("Stretch cover to fit screen")
+                            end,
+                            checked_func = function()
+                                return G_reader_settings:isTrue("screensaver_stretch_images")
+                            end,
+                            callback = function(touchmenu_instance)
+                                Screensaver:setStretchLimit(touchmenu_instance)
+                            end,
+                        },
+                },
             },
             {
-                text = _("Screensaver image"),
-                keep_menu_open = true,
-                callback = function()
-                    Screensaver:chooseFile()
-                end,
+                text = _("Unlock screen delay"),
+                sub_item_table = {
+                    genMenuItem(_("Never"), "screensaver_delay", "disable"),
+                    genMenuItem(_("1 second"), "screensaver_delay", "1"),
+                    genMenuItem(_("3 seconds"), "screensaver_delay", "3"),
+                    genMenuItem(_("5 seconds"), "screensaver_delay", "5"),
+                    genMenuItem(_("Unlock with a tap"), "screensaver_delay", "tap"),
+                    genMenuItem(_("Unlock with 'exit sleep screen' gesture"), "screensaver_delay", "gesture"),
+                },
             },
+        },
+    },
+    {
+        text = _("Sleep screen message"),
+        sub_item_table = {
             {
-                text = _("Document cover"),
-                keep_menu_open = true,
+                text = _("Add custom message to sleep screen"),
+                checked_func = function()
+                    return G_reader_settings:isTrue("screensaver_show_message")
+                end,
                 callback = function()
-                    Screensaver:chooseFile(true)
+                    G_reader_settings:toggle("screensaver_show_message")
                 end,
+                separator = true,
             },
             {
-                text = _("Screensaver message"),
-                keep_menu_open = true,
+            text = _("Edit sleep screen message"),
+            enabled_func = function() 
+                if not G_reader_settings:isTrue("screensaver_show_message") then
+                    return false    
+                end
+            end,
```

Comment:
This one will always make it disabled, as it returns `false` or `nil` (when no return or empty `return` - which evaluates to false I think - if somehow the check test for only "false" and consider "nil" as true, it's best for readability to have it here return only `true` or `false`.

Reply:
are you sure about this? it works fine on the kindle 4 I am using for testing. although will probably change it to your suggestion in the other comment.

Reply:
solved with https://github.com/koreader/koreader/pull/11549/commits/227af5c45fb4512737eb766ee6421b9be89fba53